lib/Tapable.js

```diff
@@ -2,6 +2,42 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
+
+// polyfill from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter
+// using the polyfill specifically to avoid the call to `Object.defineProperty` for performance reasons
+function fastFilter(fun/*, thisArg*/) {
```

Comment:
I don't see any difference in performance between the polyfill and the smaller function. I'll put in whichever you'd prefer and update the PR.